Source/LinqToDB/DataProvider/MySql/MySqlBulkCopy.cs

```diff
@@ -151,7 +151,14 @@ public MySqlBulkCopy(MySqlDataProvider provider)
 			// this is needed, because MySql fails on big batches, so users should be able to limit batch size
 			foreach (var batch in EnumerableHelper.Batch(source, options.MaxBatchSize ?? int.MaxValue))
 			{
+#pragma warning disable CA2000 // Dispose objects before losing scope
 				var rd = new BulkCopyReader<T>(dataConnection, columns, batch);
+#pragma warning restore CA2000 // Dispose objects before losing scope
+#if NATIVE_ASYNC
+				await using var _ = rd.ConfigureAwait(Configuration.ContinueOnCapturedContext);
+#else
+				await using var _ = rd;
+#endif
```

Comment:
I don't know that it's any better, but this is probably valid:

```suggestion
				await using var rd = new BulkCopyReader<T>(dataConnection, columns, batch)
#if NATIVE_ASYNC
					.ConfigureAwait(Configuration.ContinueOnCapturedContext);
#else
					;
#endif
```